File: back/src/main/java/csh/back/domain/trip/place/dto/response/TripPlaceFindResponse.java
package csh.back.domain.trip.place.dto.response;

import csh.back.domain.trip.place.entity.TripPlace;

public record TripPlaceFindResponse(Long placeId,
                                    String name,
                                    String address,
                                    String category,
                                    String createdBy

                                ) {
        public static TripPlaceFindResponse from(TripPlace tripPlace) {
            return new TripPlaceFindResponse(
                    tripPlace.getId(),
                    tripPlace.getName(),
                    tripPlace.getAddress(),
                    tripPlace.getTheme(),
                    tripPlace.getCreatedBy().getMember().getName()
            );
        }
    }

File: back/src/main/kotlin/csh/back/domain/trip/place/dto/response/TripPlaceFindResponse.kt
package csh.back.domain.trip.place.dto.response

import csh.back.domain.trip.place.entity.TripPlace

data class TripPlaceFindResponse(
    val tripPlaceId: Long,
    val name: String,
    val address: String,
    val category: String,
    val createdBy: String,
) {
    companion object {
        @JvmStatic
        fun from(tripPlace: TripPlace): TripPlaceFindResponse = TripPlaceFindResponse(
            tripPlaceId = requireNotNull(tripPlace.id),
            name = tripPlace.name,
            address = tripPlace.address,
            category = tripPlace.category,
            createdBy = tripPlace.createdBy.member.name,
        )
    }
}